mesonbuild/utils/universal.py

```diff
@@ -1876,10 +1873,7 @@ def detect_subprojects(spdir_name: str, current_dir: str = '',
 
 
 def substring_is_in_list(substr: str, strlist: T.List[str]) -> bool:
-    for s in strlist:
-        if substr in s:
-            return True
-    return False
+    return any(substr in s for s in strlist)
```

Comment:
IIRC these any() approaches are much slower. It's an interesting quirk of a low-level function whose body is basically entirely defined by either a short-circuiting for loop or a short-circuiting `any()`.

Reply:
The ruff `Why is this bad?` section says it is faster but perhaps you have a benchmark that proves it wrong.

% [`ruff rule SIM110`](https://docs.astral.sh/ruff/rules/reimplemented-builtin/)
# reimplemented-builtin (SIM110)

## Why is this bad?
Using a builtin function is more concise and readable. Builtins are also
more efficient than `for` loops.

---

I am OK to turn off `ruff rule SIM110` if you think that is best.  I find that when I read the code out loud, the use of `any()` makes the code read the way I think.  Others might disagree.

Reply:
It's really unfortunate because I think that `any()`/`all()` is **_MUCH_** more readable than the for loop, and the vertical space saving is very nice. However, any seems to be about 1/3 as fast (Python 3.11.9, NixOS):

```python
loop = 'abcdef'.split()
found = 'f'
nfound = 'g'


def test1():
    for x in loop:
        if x == found:
            return True
    return False


def test2():
    return any(x == found for x in loop)


def test3():
    for x in loop:
        if x == nfound:
            return True
    return False


def test4():
    return any(x == nfound for x in loop)

if __name__ == "__main__":
    import timeit

    print('for loop (found)    :', timeit.timeit(test1))
    print('for loop (not found):', timeit.timeit(test3))
    print('any() (found)       :', timeit.timeit(test2))
    print('any() (not found)   :', timeit.timeit(test4))
```

```
for loop (found)    : 0.051076093994197436
for loop (not found): 0.04388196699437685
any() (found)       : 0.15422860698890872
any() (not found)   : 0.15568504799739458
```

Reply:
On Python 3.13 release candidate 1 on M1 Mac:
```
for loop (found)    : 0.09455837499990594
for loop (not found): 0.09435533400028362
any() (found)       : 0.27777291600068565
any() (not found)   : 0.27852720800001407
```

Should we drop SIM110 ?